Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -49,4 +51,38 @@ lemma periodic {m : ℕ} (hm : n ∣ m) : Function.Periodic χ m := by
   rw [← ZMod.nat_cast_zmod_eq_zero_iff_dvd] at hm
   simp only [hm, add_zero]
 
+/-- A function that modifies the level of a Dirichlet character to some multiple
+  of its original level. -/
+noncomputable def change_level {R : Type} [CommMonoidWithZero R] {n m : ℕ} (hm : n ∣ m) :
+    DirichletCharacter R n →* DirichletCharacter R m :=
+  { toFun := fun ψ ↦ MulChar.ofUnitHom (ψ.toUnitHom.comp (ZMod.unitsMap hm)),
+    map_one' := by ext; simp,
+    map_mul' := fun ψ₁ ψ₂ ↦ by ext; simp }
+
+lemma change_level_def {m : ℕ} (hm : n ∣ m) :
+    change_level hm χ = MulChar.ofUnitHom (χ.toUnitHom.comp (ZMod.unitsMap hm)) := rfl
+
+lemma change_level_def' {m : ℕ} (hm : n ∣ m) :
+    (change_level hm χ).toUnitHom = χ.toUnitHom.comp (Units.map (ZMod.castHom hm (ZMod n))) := by
```

Comment:
Is there a reason why you are not using `ZMod.unitsMap` here?

Reply:
nope, just forgot to replace this instance of it...